### DISCUSSION TITLE
Customize options for scaleBar

### DISCUSSION THREAD

**Discussion by cyrilmarceau** [rnmapbox/maps#2983] [Ideas]
Hello,

Would it be possible to implement scalebar customization options ?

https://docs.mapbox.com/android/legacy/plugins/guides/scalebar/#customizing-the-scale-bar

Actually, only scaleBarEnabled and scaleBarPosition are developped

Thanks,
Cyril

**Comment by mfazekas** [comment]
Note that that is the old version of Mapbox not the v10 one. 
This are the docs for v10 versions:
https://docs.mapbox.com/android/maps/api/10.15.0/mapbox-maps-android/com.mapbox.maps.plugin.scalebar/-scale-bar/

It should be straightforward to implement but no concrete plan

**Reply by NathanNovak** [reply]
That link looks like it is for Android. Anything for iOS?

**Reply by NathanNovak** [reply]
Is there going to be options for units soon for iOS and Android?

**Reply by joshfendi-sce** [reply]
is there a way to shrink the width of the scale bar?

**Comment by cyrilmarceau** [comment]
Absolutely, I hadn't noticed the link.

The current problem with the default options is that the text is very small and on a map with a light background it is not very readable

**Comment by lufke** [comment]
could implement the metric units system for the units? there are just 3 o 4 countries that use imperial units system. Thanks

![image](https://github.com/rnmapbox/maps/assets/28881274/ec89a530-a965-47d4-8a55-26e7d325e84a)

**Reply by dryprogrammer** [reply]
+1 for isMetricUnits
https://docs.mapbox.com/android/maps/api/10.15.0/mapbox-maps-android/com.mapbox.maps.plugin.scalebar.generated/-scale-bar-settings/

Took a while to find this issue. I just wanted to show meters kilometers instead of miles in the scalebar. 

Does this code help?
```
@ReactMethod
fun enableMetricScaleBar() {
    val scaleBarPlugin = mapView.plugins.findPlugin(ScaleBarPluginImpl::class.java)
    scaleBarPlugin?.settings?.isMetric = true
}

import { NativeModules } from 'react-native';
const { YourCustomModule } = NativeModules;

// Call this method at the appropriate time (e.g., after map initialization)
YourCustomModule.enableMetricScaleBar();
```

I'm currently in Expo, and didn't try making plugins yet.

**Reply by dryprogrammer** [reply]
I only see 2 red dots there, where's the other 2?

**Reply by lufke** [reply]
Liberia, US and Myanmar
